### ISSUE TITLE
Unlike Narrator, NVDA does not interact correctly with the games in "Microsoft Solitaire Collection"

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#7827]
I've been watching how the accessibility in the game pack "Microsoft Solitaire Collection" progresses lately. I've noticed some good improvements lately, including high-contrast mode support. Screen readers started to interact with more controls in the user interface of the game. Unlike Narrator which interacts with the games themselves in the game pack however, NVDA fails to do so. When the system focuses on a card or another similar control in one of the games, NVDA says only "Unknown". In comparison, Narrator correctly reports information about the control (e.g. the name of the card if it is a card).

### Steps to reproduce:
1. Find and launch the game "Microsoft Solitaire Collection". If it is not installed on your system, go to the Microsoft Store, search for it there and install it from there (it is free).
2. When the game opens, find in it one of the games and launch it with ENTER.
3. Press ESCAPE to dismiss the dialog that tells you how to play the game.
4. Use TAB to navigate between controls. The "Unknown" controls start after the 3rd or the 4th TAB press.
5. Then use the arrow keys. The focus is now in the area with cards.

### Expected behavior:
NVDA should read the "unknown" controls, most of which are the cards in the games, if I'm not mistaking. The others are similar controls, associated with the game in question.

### Actual behavior:
NVDA only says "unknown" for those types of controls

### System configuration:
- Operating system: Windows 10 Pro version 1709 (build 16299.98), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: next-14685,bc898bdc, installed.
- Microsoft Solitaire Collection version: 3.18.11201.0.
- Processor: Intel Core i5-4670 at 3.40GHz.
- RAM Memory: 8.00GB.
- Graphics: Palit GeForce GT 630, 1024MB dedicated memory + 4065MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC887 at Intel Lynx Point PCH - High Definition Audio Controller.

**Comment by josephsl** [comment]
Hi, I’ll take care of it once my school finals are over. Tanks.

**Comment by Qchristensen** [comment]
Interesting, I can't get it to read ANYTHING with either Narrator or NVDA.

I am using Windows Insider Fast build 17046 and NVDA 2017.4rc4.

When I opened the Solitaire Collection, I got a message that as I am using an insider build of windows, they'd given me a preview version of the solitaire collection to try new features, so not sure if that makes a difference.  (Note I had to read that visually, as neither NVDA nor Narrator could read it).  I hadn't tried it previously, but it would be disappointing if accessibility has regressed.

The only key that I can get to work anywhere in the main menu of the game at least, is down arrow, which scrolls the window visually, but without selecting anything or changing selection.

When I go into a game (Klondike), I get a message telling me all about new solvable decks with a "Let's go" button which is pulsing.  Again, no keyboard focus and I can't dismiss it with escape, space, enter, or anything else I can think of.

Once I'm finally in the game, I can navigate with the keyboard, (arrow keys), and enter / space to select and move cards, however I can't read anything with either Narrator or NVDA.

**Comment by josephsl** [comment]
Hi, I will not proceed further unless all the puzzle pieces are here, including what build we’re dealing with. Thanks.

**Comment by josephsl** [comment]
Actually, we can proceed, now that we do have a build number.

I'll take a look at various builds to find out if the claim is true or not (the claim that Narrator can read the cards). Thanks.

**Comment by PratikP1** [comment]
A user on another list reports that his version of Solitaire does work with Narrator. Either the app group is doing A/B testing, or there are multiple preview versions out there. FYI, my version on a similar setup to Quenton, does not speak with Narrator either.

**Comment by josephsl** [comment]
Hi, mine does. It also may have to do with versions and which Insider ring (or stable build) one is in (my laptop is on release preview, while my desktop is on slow ring and my VM is on latest fast ring). Thanks.

**Comment by jfayre** [comment]
Hello,
Has there been any work on this? I just discovered that Solitaire doe work with Narrator and still does not work with NVDA.
This is on the latest Next build of NVDA and latest fast ring win10 build.
Thanks!

**Comment by Adriani90** [comment]
@josephsl did you investigate on this further?

**Comment by k-kolev1985** [comment]
@Adriani90 I'm not @josephsl, but I'll reply, since I have some information on the matter. @josephsl did indeed work on this and now NVDA speaks the cards and decks in Microsoft Solitaire Collection with the help of the "Windows 10 App Essentials" add-on. And thanks to @josephsl for doing this - it is much appreciated! 😊

**Comment by Adriani90** [comment]
@k-kolev1985 so does this mean you tested also without the addon and it is still not working properly?

**Comment by josephsl** [comment]
Hi,

Technical: one needs to ask cards for localized control type, as it comes from XAML directly, not through usual UIA property channels. One general workaround is adding support for it as part of object.roleText, but we run into issues such as making sure a given uIA element returns actual localized control type, conflict between role as given by UIA and localized control type, among other issues.

I imagine a fix from Microsoft would be nice, but given what we're dealing with (a game), it might take a while (unless people are willing to send in accessibility feedback about games to Microsoft, including calls for better UIA implementation).

Thanks.

**Comment by josephsl** [comment]
Hi,

Although not related to Solitaire directly, there is another instance where localized control type is used: Windows 11's Settings breadcrumb bar items. Narrator can read this, but you must install an add-on to get NVDA to recognize breadcrumb bar items properly. I stand by what I wrote last year: it would be better to resolve this from the source (Microsoft).

Thanks.

**Comment by josephsl** [comment]
Hi,

Resolved in Solitaire 4.13 (northern summer 2022) as NVDA can now read card sui9ts and ranks. Closing.

Thanks.
